### PR TITLE
Simplify color arguments + grid gap

### PR DESCRIPTION
**Pull request by thisisdano**
This simplifies the arguments accepted by the `color()` function, so instead of accepting either shortcodes or "triplets" (ie [family], [grade], [variant]), it now only accepts shortcodes. A new function called `advanced-color()`, based on the old `color()` function, now accepts triplets.

This simplifies both the formatting and the function itself. It saves on typing and now makes color input consistent across all functions and utility mixins:

```sass
color: color('red-50v');
@include u-text('red-50v');
$theme-color-primary-vivid: 'red-50v';
```

`advanced-color()` could be used only in certain circumstances where the specifics of something like template math could make it simpler to just use triplets: `color: advanced-color($primary-family, $adjusted-grade, $if-vivid)`

- - -

**What's included:**
- Uses color shortcodes (`'red-warm-50v'`) for color values instead of "triplets" (`'red-warm', 50, 'vivid'`)
- Updates the `color()` fn to _only_ accept shortcodes
- Uses an `advanced-color()` fn to replicate the triplet functionality of the old `color()` fn (I thought there might be some legit use of advanced color functionality)
- Uses color variables in utilities palettes to prevent running unnecessary functions
- Adds an explicit `grid-gap-md`( in addition to `grid-gap`) for consistency
- Renames `$theme-column-gap` to `$theme-column-gap-md` for consistency

- - -

**Example:**

```sass
background-color: color('red-50v');
background-color: advanced-color('red', 50, 'vivid');
@include u-bg('red-50v');

background-color: color('primary');
background-color: advanced-color('primary');
@include u-bg('primary');

background-color: color('primary-vivid');
background-color: advanced-color('primary', 'vivid');
@include u-bg('primary-vivid');

// these are equivalent
@include grid-gap
@include grid-gap('md')
```

- - -

**Next steps**
Update documentation

### CODE REVIEW

**Comment by thisisdano** [comment]
Having second thoughts about this line of work. Vars can still be very useful and too many `map-get()` feels cumbersome. I like the work here around the color fns, so I may try to keep that and revert much of the map stuff...

File: src/stylesheets/settings/_settings-color.scss [hunk 1]
```diff
@@ -18,57 +18,57 @@ Theme palette colors
...
-$theme-color-primary-darker:       'blue-warm', 80, vivid !default;
-$theme-color-primary-darkest:      false !default;
+$theme-color-primary-family:        'blue' !default;
+$theme-color-primary-lightest:      false !default;
+$theme-color-primary-lighter:       '#{$theme-color-primary-family}-10' !default;
+$theme-color-primary-light:         '#{$theme-color-primary-family}-30' !default;
+$theme-color-primary:               '#{$theme-color-primary-family}-60v' !default;
+$theme-color-primary-vivid:         'blue-warm-60v' !default;
+$theme-color-primary-dark:          'blue-warm-70v' !default;
+$theme-color-primary-darker:        'blue-warm-80v' !default;
```
Comment (maya) [inline src/stylesheets/settings/_settings-color.scss]: Why don't these have the `'#{$theme-color-primary-family}-80v'` pattern? Aren't we overriding them in the theme file? Wouldn't that be more abstracted / reuasable to keep these clean and consistent?

File: src/stylesheets/settings/_settings-color.scss [hunk 2]
```diff
@@ -18,57 +18,57 @@ Theme palette colors
...
-$theme-color-accent-cool-dark:     $theme-color-accent-cool-family, 40, vivid !default;
-$theme-color-accent-cool-darker:   'blue', 60 !default;
-$theme-color-accent-cool-darkest:  false !default;
+$theme-color-accent-cool-family:    'cyan' !default;
+$theme-color-accent-cool-lightest:  false !default;
+$theme-color-accent-cool-lighter:   '#{$theme-color-accent-cool-family}-5' !default;
+$theme-color-accent-cool-light:     '#{$theme-color-accent-cool-family}-20' !default;
+$theme-color-accent-cool:           '#{$theme-color-accent-cool-family}-30v' !default;
+$theme-color-accent-cool-dark:      '#{$theme-color-accent-cool-family}-40v' !default;
+$theme-color-accent-cool-darker:    'blue-60' !default;
```
Comment (maya) [inline src/stylesheets/settings/_settings-color.scss]: Same as above: should this be `'#{$theme-color-accent-cool-family}-60v'` and overridden in the theme file?

**Comment by maya** [comment]
Approved, we can resolve the above Q's in another PR / issue.

**Pull request merged**